middleware/monitor/monitor.go

```diff
@@ -84,6 +84,7 @@ func New(config ...Config) fiber.Handler {
 		}
 		if c.Get(fiber.HeaderAccept) == fiber.MIMEApplicationJSON || cfg.APIOnly {
 			mutex.Lock()
+			defer mutex.Unlock()
```

Comment:
why do we close the mutext so late? after writing the status and the json? 
besides, in previous benchmarks it had turned out that the defer calls are slower

Reply:
Otherwise line 97 (`return c.Status(fiber.StatusOK).JSON(data)`) has a race condition where it reads the shared `data` variable.

True, `defer`-ed functions run a tad slower, but since one of the more recent Go releases it's negligible (and will probably be reduced further in the future).

Instead of using `defer` here, we could also make a deep copy of `data` while still having the mutex locked.

Reply:
ok it can stay